### PR TITLE
ref(ui): Convert dateSummary to function, use RTL

### PR DESCRIPTION
**Pull request by scttcper**

### CODE REVIEW

File: tests/js/spec/components/organizations/timeRangeSelector/dateSummary.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import {mountWithTheme, screen} from 'sentry-test/reactTestingLibrary';
+
+import DateSummary from 'app/components/organizations/timeRangeSelector/dateSummary';
+
+const start = new Date('2017-10-14T02:38:00.000Z');
+const end = new Date('2017-10-17T02:38:00.000Z'); // National Pasta Day
+
+describe('DateSummary', function () {
+  const getComponent = (props = {}) => <DateSummary start={start} end={end} {...props} />;
+  const createWrapper = (props = {}) => mountWithTheme(getComponent(props));
```
[the remark this answers is not in the document]
Comment (scttcper) [reply tests/js/spec/components/organizations/timeRangeSelector/dateSummary.spec.tsx]: fixed

File: tests/js/spec/components/organizations/timeRangeSelector/dateSummary.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import {cleanup, mountWithTheme, screen} from 'sentry-test/reactTestingLibrary';
+
+import DateSummary from 'app/components/organizations/timeRangeSelector/dateSummary';
+
+const start = new Date('2017-10-14T02:38:00.000Z');
+const end = new Date('2017-10-17T02:38:00.000Z'); // National Pasta Day
```
Comment (priscilawebdev) [inline tests/js/spec/components/organizations/timeRangeSelector/dateSummary.spec.tsx:6]: `// National Pasta Day` lol

File: tests/js/spec/components/organizations/timeRangeSelector/dateSummary.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  it('renders', async () => {
+    const {container} = mountWithTheme(<DateSummary start={start} end={end} />);
+    expect(container).toSnapshot();
+  });
+
+  it('does not show times when it is midnight for start date and 23:59:59 for end date', () => {
+    mountWithTheme(<DateSummary start={start} end={end} />);
+    // Search by year because day may change depending on timezone
+    expect(screen.getAllByText(/Oct.+2017/)[0].childElementCount).toBe(1);
+    cleanup();
```
Comment (priscilawebdev) [inline tests/js/spec/components/organizations/timeRangeSelector/dateSummary.spec.tsx:18]: should we use [rerender](https://testing-library.com/docs/react-testing-library/api/#rerender) instead?

File: tests/js/spec/components/organizations/timeRangeSelector/dateSummary.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import {cleanup, mountWithTheme, screen} from 'sentry-test/reactTestingLibrary';
+
+import DateSummary from 'app/components/organizations/timeRangeSelector/dateSummary';
+
+const start = new Date('2017-10-14T02:38:00.000Z');
+const end = new Date('2017-10-17T02:38:00.000Z'); // National Pasta Day
+
+describe('DateSummary', () => {
+  it('renders', async () => {
```
Comment (priscilawebdev) [inline tests/js/spec/components/organizations/timeRangeSelector/dateSummary.spec.tsx:9]:
```suggestion
  it('renders', () => {
```

**Review by matejminar — APPROVED** [review]
Looks good to me with Priscila's comments.

**Pull request merged**